### PR TITLE
Add the colors parameter and radius scale 

### LINKED ISSUE #150: Attractor Radius in StaticGeodesicPlotter is too less
🐞 **Problem**

Due to the radius issues, the plots are not shown properly. Most of the times, they miss an Attractor.

🎯 **Goal**

Think of a strategy to get good plots!

💡 **Possible solutions**

📋  **Steps to solve the problem**

 * Comment below about what you've started working on.
 * Add, commit, push your changes
 * Submit a pull request and add this in comments - `Addresses #<put issue number here>`
 * Ask for a review in comments section of pull request
 * Celebrate your contribution to this project 🎉

### LINKED ISSUE #156: Color options in ScatterGeodesicPlotter and StaticGeodesicPlotter is missing.
🐞 **What bugs you?**
Color options for attractor and scatter plots are not present in the classes ScatterGeodesicPlotter and StaticGeodesicPlotter.

🎯 **Goal**
This change can benefit people in the way that if they want to create many plots, It can help them in distinguishing a certain plot in a group of plots. Also it gives user to option to use his/her favorite color.

💡 **Possible solutions**
The possible solution is to use a parameter for the user to pass the color while making the class. If the user does not want to pass the parameter, we can default it to some color.

📋  **Steps to solve the problem**
I have created and tested the class with added parameters. Will create a PR after writing this issue.
 * Comment below about what you've started working on.
 * Add, commit, push your changes
 * Submit a pull request and add this in comments - `Addresses #<put issue number here>`
 * Ask for a review in comments section of pull request
 * Celebrate your contribution to this project 🎉

### PR DESCRIPTION
**Pull request by rtanmay**
Fixes #156  Fixes #150 

I have added the radius scaling factor (defaults to 1) which will scale the attractor radius, the picture below shows the results.
Screenshot from 2019-04-07 17-16-53

I have added the color parameters in ScatterGeodesicPlotter classes which defaluts to :
attractor_color= "black" and cmap_color="Oranges" in ScatterGeodesicPlotter class
This is how it looks:
Screenshot from 2019-04-07 17-12-16

Color Attribute in StaticGeodesicPlotter:
Screenshot from 2019-04-07 19-05-22

### CODE REVIEW

**Comment by shreyasbapat** [comment]
You have to get the quality checks passed first. I will review the PR thoroughly after that! :)

**Comment by shreyasbapat** [comment]
The main aim of #150 was that the way the plots are made must be pleasing. As in your pull request, you can see that the attractor has a very large radius, which is wrong. There should be some strategy to keep radius of object according to the plot!

**Comment by rtanmay** [comment]
@shreyasbapat  Alright will fix the radius dynamically according to the dimensions of the plot. I think having a manual scaling option also will be good in case the user wants to adjust it to his/her needs.

**Comment by shreyasbapat** [comment]
> I think having a manual scaling option also will be good in case the user wants to adjust it to his/her needs.

Totally agreed!

**Comment by shreyasbapat** [comment]
@rtanmay Updates?

**Comment by rtanmay** [comment]
@shreyasbapat  I have written the method of auto scaling but there is a bug in that. It's almost done. Will create a PR sometime around 6AM.

**Comment by rtanmay** [comment]
@shreyasbapat are these proportions good?
Screenshot from 2019-04-11 01-43-09

**Comment by rtanmay** [comment]
All done now. This is how it looks:
Screenshot from 2019-04-11 04-12-31

can you check @shreyasbapat now?
Thanks

**Comment by shreyasbapat** [comment]
@rtanmay Do you see the attractor going beyond the orbit? That shouldn't happen

**Comment by Varunvaruns9** [comment]
@rtanmay Any updates on this? Its a blocker for #189 .

**Comment by rtanmay** [comment]
@shreyasbapat Is this fine now?
Screenshot from 2019-04-13 03-41-54
Screenshot from 2019-04-13 03-41-30

@Varunvaruns9 If shreyas says its fine now then I will just write the test for code coverage to clear them. Won't take much time.

**Comment by shreyasbapat** [comment]
Can you plot earth once?

**Comment by rtanmay** [comment]
@shreyasbapat 

Earth around sun
Screenshot from 2019-04-13 14-50-32

File: src/einsteinpy/plotting/geodesics_static.py [hunk 1]
```diff
@@ -45,10 +133,37 @@ def plot_attractor(self):
         if not self._attractor_present:
             self._draw_attractor()
 
-    def _draw_attractor(self, min_radius=10 * u.km):
-        radius = max(schwarzschild_radius(self.mass) * 10000, min_radius.to(u.km))
-        color = "#ffcc00"
-        self.ax.add_patch(mpl.patches.Circle((0, 0), radius.value, lw=0, color=color))
+    def get_x_y(self, pos_vec, vel_vec, end_lambda, step_size):
```
Comment (shreyasbapat) [inline src/einsteinpy/plotting/geodesics_static.py]: Private function please

**Comment by shreyasbapat** [comment]
Can you rebase once? Just ignore the @pep8speaks  Comment for now :)

**Comment by rtanmay** [comment]
@shreyasbapat  Yes, will do that after clearing the coverage checks once.

**Comment by rtanmay** [comment]
Animate checks are failing after rebasing.
Screenshot from 2019-04-15 20-05-19

I tried adding mock patch since I thought it was not there but even that didn't solve the problem.
Can you check? @Varunvaruns9 @shreyasbapat

File: src/einsteinpy/plotting/geodesics_static.py [hunk 2]
```diff
@@ -11,16 +13,67 @@
...
+    ----------
+    attractor_radius_scale :
+        Scales the attractor radius by the 
+        value given. Default is 1. It is used to 
+        make plots look more clear if needed.
+    attractor_color : 
+        Color which is used to denote the attractor.
+        Defaults to #ffcc00.
+
+    Functions and Class Variables
```
Comment (shreyasbapat) [inline src/einsteinpy/plotting/geodesics_static.py]: No such section in docstrings  :)

File: src/einsteinpy/plotting/geodesics_scatter.py [hunk 3]
```diff
@@ -4,21 +4,40 @@
 from matplotlib.animation import FuncAnimation
 
 from einsteinpy.metric import Schwarzschild
+from einsteinpy.utils import schwarzschild_radius
 
 
 class ScatterGeodesicPlotter:
     """
     Class for plotting static matplotlib plots.
+
```
Comment (shreyasbapat) [inline src/einsteinpy/plotting/geodesics_scatter.py]: Why are parameters specified in class docstring?

File: src/einsteinpy/plotting/geodesics_scatter.py [hunk 4]
```diff
@@ -4,21 +4,40 @@
...
+    Parameters
+    ----------
+    attractor_color : 
+        Color which is used to denote the attractor.
+        Defaults to black.
+    cmap_color : 
+        Color used in function plot.
+
+    Functions
```
Comment (shreyasbapat) [inline src/einsteinpy/plotting/geodesics_scatter.py]: No functions required

File: src/einsteinpy/plotting/geodesics_scatter.py [hunk 5]
```diff
@@ -114,3 +133,6 @@ def show(self):
 
     def save(self, name="scatter_geodesic.png"):
         plt.savefig(name)
+
+    def get_swcradius(self):
```
Comment (shreyasbapat) [inline src/einsteinpy/plotting/geodesics_scatter.py]: I think we don't need to add a redundant function

File: src/einsteinpy/plotting/geodesics_static.py [hunk 6]
```diff
@@ -11,16 +13,67 @@
 
 class StaticGeodesicPlotter:
     """
-    Class for plotting static matplotlib plots
+    Class for plotting static matplotlib plots.
+
```
Comment (shreyasbapat) [inline src/einsteinpy/plotting/geodesics_static.py]: Please add this documentation in rst format in plotting docs.

File: src/einsteinpy/plotting/geodesics_static.py [hunk 7]
```diff
@@ -11,16 +13,67 @@
...
+    ):
         self.ax = ax
         if not self.ax:
             _, self.ax = plt.subplots(figsize=(6, 6))
         self.time = time
         self.mass = mass
         self._attractor_present = False
+        self.attractor_color = attractor_color
+        self.attractor_radius_scale = attractor_radius_scale
+        self.xarr = False
```
Comment (shreyasbapat) [inline src/einsteinpy/plotting/geodesics_static.py]: This should be a list initially?

Comment (shreyasbapat) [reply src/einsteinpy/plotting/geodesics_static.py]: Also private

File: src/einsteinpy/plotting/geodesics_static.py [hunk 8]
```diff
@@ -11,16 +13,67 @@
...
+        self.yarr = False
+        self.get_curr_plot_radius = 0
```
Comment (shreyasbapat) [inline src/einsteinpy/plotting/geodesics_static.py]: What is this for?

File: src/einsteinpy/plotting/geodesics_static.py [hunk 9]
```diff
@@ -45,10 +98,70 @@ def plot_attractor(self):
...
+        self.xarr = x
+        self.yarr = y
+        return x, y
+
+    def mindist(self, x, y):
+        return math.sqrt(x * x + y * y)
+
+    def _draw_attractor(self):
+        """ 
+            Auto scaling according to plots
```
Comment (shreyasbapat) [inline src/einsteinpy/plotting/geodesics_static.py]: Indentation. Plus wrong use of docstrings.

Have a look at numpy like docstrings

File: src/einsteinpy/tests/test_plotting/test_scattergeodesicplotter.py [hunk 10]
```diff
@@ -8,7 +8,10 @@
 
 
 @pytest.fixture()
-def dummy_data():
+def create_test_datas():
+    """
```
Comment (shreyasbapat) [inline src/einsteinpy/tests/test_plotting/test_scattergeodesicplotter.py]: Why docstring here?

File: src/einsteinpy/tests/test_plotting/test_scattergeodesicplotter.py [hunk 11]
```diff
@@ -8,7 +8,10 @@
 
 
 @pytest.fixture()
-def dummy_data():
+def create_test_datas():
```
Comment (shreyasbapat) [inline src/einsteinpy/tests/test_plotting/test_scattergeodesicplotter.py]: Now that I think of it, dummy data is better

File: src/einsteinpy/tests/test_plotting/test_scattergeodesicplotter.py [hunk 12]
```diff
@@ -19,8 +22,11 @@ def dummy_data():
     return r, v, t, m, start_lambda, end_lambda, step_size
 
 
-def test_plot_attractor_is_called_only_once(dummy_data):
-    r, v, _, m, _, el, ss = dummy_data
+def test_plot_attractor_is_called_only_onces(create_test_datas):
+    """
```
Comment (shreyasbapat) [inline src/einsteinpy/tests/test_plotting/test_scattergeodesicplotter.py]: No docstrings here

File: src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py [hunk 13]
```diff
@@ -21,29 +24,39 @@ def dummy_data():
...
     ss = 0.5e-6
     cl = StaticGeodesicPlotter(m)
     cl.plot(r, v, el, ss)
+    cl.show()
     mock_plot_attractor.assert_called_with()
     assert cl._attractor_present
 
 
-@mock.patch("einsteinpy.plotting.geodesics_static.plt.savefig")
+@mock.patch("einsteinpy.plotting.geodesics_static.StaticGeodesicPlotter.save")
```
Comment (shreyasbapat) [inline src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py]: Why did you change this?

File: src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py [hunk 14]
```diff
@@ -54,3 +67,121 @@ def test_plot_save_saves_plot(mock_save):
...
+    ss = 0.5e-6
+    cl = StaticGeodesicPlotter(m, attractor_radius_scale=1500)
+    cl.plot(r, v, el, ss)
+    mock_plot_attractor.assert_called_with()
+    assert cl._attractor_present
+    assert cl.attractor_radius_scale == 1500
+
+
+@mock.patch("einsteinpy.plotting.geodesics_static.StaticGeodesicPlotter.plot_attractor")
+def test_plot_calls_plot_attractor3(mock_plot_attractor):
```
Comment (shreyasbapat) [inline src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py]: I see you have written same test multiple times. Why?

File: src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py [hunk 15]
```diff
@@ -54,3 +67,121 @@ def test_plot_save_saves_plot(mock_save):
...
+    Test for checking the attractor_radius_scale passed
+    """
+    r = [306 * u.m, np.pi / 2 * u.rad, np.pi / 2 * u.rad]
+    v = [0 * u.m / u.s, 0 * u.rad / u.s, 951.0 * u.rad / u.s]
+    m = 4e24 * u.kg
+    el = 0.002
+    ss = 0.5e-6
+    cl = StaticGeodesicPlotter(m, attractor_radius_scale=1500)
+    cl.plot(r, v, el, ss)
+    mock_plot_attractor.assert_called_with()
```
Comment (shreyasbapat) [inline src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py]: Why mocking here?

File: src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py [hunk 16]
```diff
@@ -54,3 +67,121 @@ def test_plot_save_saves_plot(mock_save):
...
+    ss = 0.5e-6
+    cl = StaticGeodesicPlotter(m, attractor_color="blue")
+    cl.plot(r, v, el, ss)
+    mock_plot_attractor.assert_called_with()
+    assert cl._attractor_present
+    assert cl.attractor_color == "blue"
+
+
+@mock.patch("einsteinpy.plotting.geodesics_static.StaticGeodesicPlotter.get_swcradius")
+def test_plot_get_swcradius(mock_get_swcradius):
```
Comment (shreyasbapat) [inline src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py]: What is the need of such a function?

File: src/einsteinpy/tests/test_plotting/test_scattergeodesicplotter.py [hunk 17]
```diff
@@ -30,7 +36,10 @@ def test_plot_attractor_is_called_only_once(dummy_data):
 @mock.patch(
     "einsteinpy.plotting.geodesics_scatter.ScatterGeodesicPlotter._plot_attractor"
 )
-def test_plot_calls_plot_attractor(mock_plot_attractor):
+def test_plot_calls_plot_attractor1s(mock_plot_attractor):
```
Comment (shreyasbapat) [inline src/einsteinpy/tests/test_plotting/test_scattergeodesicplotter.py]: Why renaming?

File: src/einsteinpy/plotting/geodesics_static.py (hunk 7 above)
Comment (rtanmay) [reply src/einsteinpy/plotting/geodesics_static.py]: Yes, it should be a list but datatypes can be automatically changed in python so I don't think it should be a problem, I will initialize it to empty list anyway.

File: src/einsteinpy/tests/test_plotting/test_scattergeodesicplotter.py (hunk 17 above)
Comment (rtanmay) [reply src/einsteinpy/tests/test_plotting/test_scattergeodesicplotter.py]: I was testing for the coverage, which functions are not executing, so I added 1,2,3 after the names so it will be easy to check where it happened. I planned to change this once the checks were passed.

File: src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py (hunk 16 above)
Comment (rtanmay) [reply src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py]: I made a function, get_swcradius for getting the Schwarzschild radius, so I was testing it otherwise it may fail the circleci/patch.

File: src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py (hunk 14 above)
Comment (rtanmay) [reply src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py]: Multiple tests are for different functions made in the staticgeodesicplotter. Just named them 1,2,3 in hurry to see if it passes the checks

File: src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py (hunk 13 above)
Comment (rtanmay) [reply src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py]: Oh, this. I thought since some others are starting from StaticGeodesicPlotter, there should be uniformity?

File: src/einsteinpy/tests/test_plotting/test_scattergeodesicplotter.py (hunk 12 above)
Comment (rtanmay) [reply src/einsteinpy/tests/test_plotting/test_scattergeodesicplotter.py]: Okay

File: src/einsteinpy/plotting/geodesics_static.py (hunk 8 above)
Comment (rtanmay) [reply src/einsteinpy/plotting/geodesics_static.py]: Okay, so for autoscaling, the radius is made the minimum of the minimum distance and 1/12th the dimensions of plot.

However, when using manual scaling, the user won't know what number to put in the attractor_radius_scale, In the example: I used 1500 but that is by trial and error and the user will have to run the program again to find the suitable number. 

Now, with get_curr_plot_radius, we get the radius of the attractor as plotted in the plot. this radius can be divided with the get_curr_plot_radius in auto scaling to get the multiplier which otherwise would take many attempts at trial and error, plus the program will have to executed several times.

The xarr and yarr are the plotting points of the particle moving around the attractor. I need these to calculate the minimum distance between the particle and attractor. But these are not available to me since these are in the function which is finished before calling the plot attractor function. That's why I had to make these class members.

File: src/einsteinpy/tests/test_plotting/test_scattergeodesicplotter.py (hunk 10 above)
Comment (rtanmay) [reply src/einsteinpy/tests/test_plotting/test_scattergeodesicplotter.py]: I thought I had to document the test functions I made?

File: src/einsteinpy/tests/test_plotting/test_scattergeodesicplotter.py (hunk 11 above)
Comment (rtanmay) [reply src/einsteinpy/tests/test_plotting/test_scattergeodesicplotter.py]: Yeah, will change

File: src/einsteinpy/plotting/geodesics_scatter.py (hunk 3 above)
Comment (rtanmay) [reply src/einsteinpy/plotting/geodesics_scatter.py]: @ I didn't know it had to be in the rst. will shift them

File: src/einsteinpy/plotting/geodesics_scatter.py (hunk 4 above)
Comment (rtanmay) [reply src/einsteinpy/plotting/geodesics_scatter.py]: Okay

File: src/einsteinpy/plotting/geodesics_scatter.py (hunk 5 above)
Comment (rtanmay) [reply src/einsteinpy/plotting/geodesics_scatter.py]: Won't this be necessary for manual scaling? To get the multiplier.

I thought this function should be there for the geodesicplotterclass object otherwise the user will have to call it separately.
Will remove this if you say so.

File: src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py (hunk 16 above)
Comment (shreyasbapat) [reply src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py]: Why do we need that function? That's my point :sweat_smile:

File: src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py (hunk 13 above)
Comment (shreyasbapat) [reply src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py]: Read about what mocking is :) 
https://docs.python.org/3/library/unittest.mock.html

File: src/einsteinpy/plotting/geodesics_static.py (hunk 7 above)
Comment (shreyasbapat) [reply src/einsteinpy/plotting/geodesics_static.py]: Its not a problem, but the code should not be misleading

File: src/einsteinpy/plotting/geodesics_scatter.py (hunk 5 above)
Comment (shreyasbapat) [reply src/einsteinpy/plotting/geodesics_scatter.py]: We already have schwarzschild radius in the utility. We don't need this redundant function

File: src/einsteinpy/plotting/geodesics_scatter.py (hunk 3 above)
Comment (shreyasbapat) [reply src/einsteinpy/plotting/geodesics_scatter.py]: No, I never said checks fail due to this :)

**Comment by shreyasbapat** [comment]
Hi @rtanmay are you still working on this? Remember #177 depends on this!

**Comment by rtanmay** [comment]
@shreyasbapat  Yes, I am working on this. Was really busy for the past 2 days. Will fix it by today's night or by early Saturday.

File: src/einsteinpy/plotting/geodesics_scatter.py (hunk 4 above)
Comment (rtanmay) [reply src/einsteinpy/plotting/geodesics_scatter.py]: Done

File: src/einsteinpy/plotting/geodesics_scatter.py (hunk 5 above)
Comment (rtanmay) [reply src/einsteinpy/plotting/geodesics_scatter.py]: Okay, removed this.
Done

File: src/einsteinpy/plotting/geodesics_static.py (hunk 7 above)
Comment (rtanmay) [reply src/einsteinpy/plotting/geodesics_static.py]: Done, initialised to empty numpy array

File: src/einsteinpy/tests/test_plotting/test_scattergeodesicplotter.py (hunk 12 above)
Comment (rtanmay) [reply src/einsteinpy/tests/test_plotting/test_scattergeodesicplotter.py]: Removed them

**Comment by rtanmay** [comment]
While rebasing, something went wrong. Other's commits are also coming.
@shreyasbapat Don't review the PR now, I will let you know once the above change is reverted.

**Comment by rtanmay** [comment]
@shreyasbapat 
I think all the changes are done now. Can you check? Thanks

File: src/einsteinpy/plotting/geodesics_scatter.py [hunk 18]
```diff
@@ -9,16 +9,28 @@
 class ScatterGeodesicPlotter:
     """
     Class for plotting static matplotlib plots.
+
+    Parameters
```
Comment (shreyasbapat) [inline src/einsteinpy/plotting/geodesics_scatter.py]: Again, why does the class docstring contains this?

File: src/einsteinpy/plotting/geodesics_static.py (hunk 6 above)
Comment (shreyasbapat) [reply src/einsteinpy/plotting/geodesics_static.py]: Same here

File: src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py [hunk 19]
```diff
@@ -54,3 +55,32 @@ def test_plot_save_saves_plot(mock_save):
     name = "test_plot.png"
     cl.save(name)
     mock_save.assert_called_with(name)
+
+
+@mock.patch("einsteinpy.plotting.geodesics_static.StaticGeodesicPlotter.plot_attractor")
```
Comment (shreyasbapat) [inline src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py]: If you just have to check that every thing works fine. Why are you mocking plot attractor?

File: src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py [hunk 20]
```diff
@@ -54,3 +55,32 @@ def test_plot_save_saves_plot(mock_save):
...
+    ss = 0.5e-6
+    cl = StaticGeodesicPlotter(m, attractor_radius_scale=1500)
+    cl.plot(r, v, el, ss)
+    mock_plot_attractor_Manual.assert_called_with()
+    assert cl._attractor_present
+    assert cl.attractor_radius_scale == 1500
+    assert cl.get_curr_plot_radius != -1
+
+
+@mock.patch("einsteinpy.plotting.geodesics_static.StaticGeodesicPlotter.plot_attractor")
```
Comment (shreyasbapat) [inline src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py]: Same here

File: docs/source/api/plotting/geodesics_scatter.rst [hunk 21]
```diff
@@ -0,0 +1,31 @@
...
+    User can give the color to attractor of his/her choice.
+    It can be passed while making the object of geodesics_static class.
+    Default color of attractor is "black".
+
+    .. code-block:: python
+        
+        self.attractor_color = attractor_color
+        plt.scatter(0, 0, color=self.attractor_color)
+
+* Orbit :
```
Comment (shreyasbapat) [inline docs/source/api/plotting/geodesics_scatter.rst]: Geodesic and not Orbit :)

File: src/einsteinpy/plotting/geodesics_static.py [hunk 22]
```diff
@@ -1,4 +1,6 @@
+import math
```
Comment (shreyasbapat) [inline src/einsteinpy/plotting/geodesics_static.py]: It is better to stay in numpy ecosystem :)

File: src/einsteinpy/plotting/geodesics_scatter.py (hunk 18 above)
Comment (rtanmay) [reply src/einsteinpy/plotting/geodesics_scatter.py]: I have shifted them to the __init__ docstring. If this is also wrong. Then it would be really helpful to know where to put them.

File: docs/source/api/plotting/geodesics_scatter.rst (hunk 21 above)
Comment (rtanmay) [reply docs/source/api/plotting/geodesics_scatter.rst]: Okay, Corrected.

File: src/einsteinpy/plotting/geodesics_static.py (hunk 22 above)
Comment (rtanmay) [reply src/einsteinpy/plotting/geodesics_static.py]: Yeah, Done :)

File: src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py (hunk 19 above)
Comment (rtanmay) [reply src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py]: I am mocking plot attractor two times, the reasons are mentioned below:

1)Auto scaling
Here the _draw_attractor will be called and the "if" block will be executed and not the else block.

Now since the else block is not executed till now (because attractor_radius_scale == -1.0) I will have to call it again.

2)Manual scaling
Here the _draw_attractor will be called and the "else" block will be executed .

I don't understand why I do not have to mock plot attractor. If I don't then the lines won't be executed otherwise and the coverage will go down.

Actually I need the _draw_attractor only so I have change plot_attractor to _draw_attractor.

Can you please explain how the else block will be executed if I do not pass manual scale parameter? I don't understand how I can check that everything works fine without it.

File: src/einsteinpy/plotting/geodesics_scatter.py (hunk 18 above)
Comment (rtanmay) [reply src/einsteinpy/plotting/geodesics_scatter.py]: Also, there was no docstring for plot_trajectory so I have added them. Do let me know if it was unnecessary.

File: src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py (hunk 19 above)
Comment (shreyasbapat) [reply src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py]: Do you want to assert if `_draw_attractor` is being called or not?

Comment (shreyasbapat) [reply src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py]: Or you just want the _draw_attractor to be called?

**Comment by shreyasbapat** [comment]
Also, I would ask you to restrict the commits to <10 or as you already know git, a number which seems reasonable to you :)

File: src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py (hunk 19 above)
Comment (rtanmay) [reply src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py]: I just want the _draw_attractor to be called

Comment (rtanmay) [reply src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py]: and then I am checking if it is called or not

**Comment by rtanmay** [comment]
Yes, I will fixup the commits once I know the PR is finally ready.

File: src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py (hunk 19 above)
Comment (shreyasbapat) [reply src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py]: Well that's what I am emphasising. 

> I am checking if it is called or not

Why do you need to check that?

Comment (rtanmay) [reply src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py]: I see, so I don't really have to check that since it is being checked by other functions.
Will remove the assert line

Comment (rtanmay) [reply src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py]: Done, is it fine now?

Comment (shreyasbapat) [reply src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py]: What did you change? xD

File: src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py [hunk 23]
```diff
@@ -54,3 +55,34 @@ def test_plot_save_saves_plot(mock_save):
...
+    el = 0.002
+    ss = 0.5e-6
+    cl = StaticGeodesicPlotter(m, attractor_radius_scale=1500)
+    cl.plot(r, v, el, ss)
+    assert cl._attractor_present
+    assert cl.attractor_radius_scale == 1500
+    assert cl.get_curr_plot_radius != -1
+
+
+@mock.patch(
```
Comment (shreyasbapat) [inline src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py]: Remove this mock too!

File: src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py [hunk 24]
```diff
@@ -54,3 +55,34 @@ def test_plot_save_saves_plot(mock_save):
     name = "test_plot.png"
     cl.save(name)
     mock_save.assert_called_with(name)
+
+
+@mock.patch(
```
Comment (shreyasbapat) [inline src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py]: Remove this mock!

**Comment by shreyasbapat** [comment]
This will also need a rebase!

File: src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py (hunk 24 above)
Comment (shreyasbapat) [reply src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py]: Only mock was to be removed, not the test !

**Comment by rtanmay** [comment]
> Also, I would ask you to restrict the commits to <10 or as you already know git, a number which seems reasonable to you :)

Okay, I think 1 commit per PR is good enough :)

**Comment by shreyasbapat** [comment]
You can write a test without mocking!

File: src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py (hunk 24 above)
Comment (rtanmay) [reply src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py]: Done

File: src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py (hunk 23 above)
Comment (rtanmay) [reply src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py]: Done

**Comment by shreyasbapat** [comment]
> Okay, I think 1 commit per PR is good enough :)

I can allow upto 10

File: src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py [hunk 25]
```diff
@@ -54,3 +55,20 @@ def test_plot_save_saves_plot(mock_save):
     name = "test_plot.png"
     cl.save(name)
     mock_save.assert_called_with(name)
+
+
+def test_plot_calls_draw_attractor_Manualscale(dummy_data):
```
Comment (shreyasbapat) [inline src/einsteinpy/tests/test_plotting/test_staticgeodesicplotter.py]: Good :)

**Comment by shreyasbapat** [comment]
DeepinScreenshot_select-area_20190421025103
You didn't rebase properly!

**Comment by rtanmay** [comment]
> DeepinScreenshot_select-area_20190421025103
> You didn't rebase properly!

Done

**Comment by shreyasbapat** [comment]
Done! Thanks @rtanmay  :rocket: 
Hope to see next contribution soon.
Key takeaways:
1) Do only what's expected in the issue. Don't add unnecessary doctrings etc everywhere. This increases the diff and the reviews aren't done properly! 
2) Write tests smartly!
Amazing work :)

**Comment by rtanmay** [comment]
> Done! Thanks @rtanmay rocket
> Hope to see next contribution soon.
> Key takeaways:
> 
>     1. Do only what's expected in the issue. Don't add unnecessary doctrings etc everywhere. This increases the diff and the reviews aren't done properly!
> 
>     2. Write tests smartly!
>        Amazing work :)

Thanks :)
